src/Tizen.Uix.VoiceControlManager/Tizen.Uix.VoiceControlManager/VoiceControlManagerClient.cs

```diff
@@ -0,0 +1,1929 @@
+        /// 1) This exception can be due to an invalid state.
+        /// 2) This exception can be due to operation failed.
+        /// 3) This exception can be due to out of memory.
+        /// </exception>
+        public static void SetSelectedResults(IEnumerable<VoiceCommand> list)
+        {
+            VoiceCommandsGroup cmdList = new VoiceCommandsGroup();
+            foreach(VoiceCommand command in list)
+            {
+                cmdList.Commands.Add(command);
+            }
+            ErrorCode error = VcMgrSetSelectedResults(cmdList._handle);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "SetSelectedResults Failed with error " + error);
+                throw ExceptionFactory.CreateException(error);
+            }
+        }
+
+        /// <summary>
+        /// Sets a callback function for getting recognition result.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <privilege>http://tizen.org/privilege/voicecontrol.manager</privilege>
+        /// <privlevel>public</privlevel>
+        /// <feature>http://tizen.org/feature/speech.control</feature>
+        /// <feature>http://tizen.org/feature/microphone</feature>
+        /// <param name="callback">The callback function.
+        /// Called when client gets the all recognition results from voice control service.
+        /// Out parameters:
+        /// a = event : The result event
+        /// b = commandList : Voice control command list
+        /// c = result : Command text
+        /// d = message : Engine message
+        /// </param>
+        /// <exception cref="ArgumentException">This exception can be due to an invalid parameter.</exception>
+        /// <exception cref="InvalidOperationException">This exception can be due to an invalid state.</exception>
+        /// <exception cref="UnauthorizedAccessException">This exception can be due to permission denied.</exception>
+        /// <exception cref="NotSupportedException">This exception can be due to not supported.</exception> 
+        public static void SetRecognizedCommandsSelectionDelegate(BoolAction<ResultEvent, IEnumerable<VoiceCommand>, string, string> callback)
+        {
+            if (callback == null)
+            {
+                Log.Error(LogTag, "callback is null");
+                throw ExceptionFactory.CreateException(ErrorCode.InvalidParameter);
+            }
+
+            _allResultDel = callback;
+            _allResultDelegate = (ResultEvent evt, IntPtr vcCmdList, IntPtr result, IntPtr msg, IntPtr userData) =>
+            {
+                string resultString = Marshal.PtrToStringAnsi(result);
+                string msgString = Marshal.PtrToStringAnsi(msg);
+                VoiceCommandsGroup cmdList = new VoiceCommandsGroup(new SafeCommandListHandle(vcCmdList));
+
+                return _allResultDel.Invoke(evt, cmdList.Commands, resultString, msgString);
```

Comment:
What is evt and why need to pass into user callback?

Reply:
evt is command matching result from engine.
so VC manager developer can check Success or Rejected.
Success - it's command matching result for utterance.
Rejected - it's no command matching result for utterance.

```
    public enum ResultEvent
    {
        /// <summary>
        /// Normal result.
        /// </summary>
        Success = 0,
        /// <summary>
        /// Rejected result.
        /// </summary>
        Rejected = 1
    };
```

Reply:
Changed to RecognizedResult